shelley-ma/impl/src/Cardano/Ledger/ShelleyMA/TxBody.hs

```diff
@@ -0,0 +1,215 @@
+{-# LANGUAGE ConstraintKinds #-}
+{-# LANGUAGE DataKinds #-}
+{-# LANGUAGE DeriveAnyClass #-}
+{-# LANGUAGE DeriveGeneric #-}
+{-# LANGUAGE DerivingStrategies #-}
+{-# LANGUAGE DerivingVia #-}
+{-# LANGUAGE FlexibleContexts #-}
+{-# LANGUAGE FlexibleInstances #-}
+{-# LANGUAGE FunctionalDependencies #-}
+{-# LANGUAGE GADTs #-}
+{-# LANGUAGE GeneralizedNewtypeDeriving #-}
+{-# LANGUAGE MultiParamTypeClasses #-}
+{-# LANGUAGE NamedFieldPuns #-}
+{-# LANGUAGE OverloadedStrings #-}
+{-# LANGUAGE PatternSynonyms #-}
+{-# LANGUAGE ScopedTypeVariables #-}
+{-# LANGUAGE StandaloneDeriving #-}
+{-# LANGUAGE TypeApplications #-}
+{-# LANGUAGE TypeFamilies #-}
+{-# LANGUAGE UndecidableInstances #-}
+{-# LANGUAGE ViewPatterns #-}
+
+module Cardano.Ledger.ShelleyMA.TxBody
+  ( MaryBody (MaryBody, STxBody),
+    MaryBody' (..),
+  )
+where
+
+import Cardano.Binary (Annotator, FromCBOR (..), ToCBOR (..))
+-- \/ This is the type family Value
+import Cardano.Ledger.Core (CompactForm (..), Compactible (..), Value)
+import Cardano.Ledger.Era (Era)
+import Cardano.Ledger.ShelleyMA.Timelocks (ValidityInterval (..), decodeVI, encodeVI)
+import Data.Coders
+  ( Decode (..),
+    Encode (..),
+    decode,
+    decodeStrictSeq,
+    (!>),
+    (<!),
+  )
+import Data.MemoBytes (Mem, MemoBytes (..), memoBytes)
+import Data.Sequence.Strict (StrictSeq)
+import Data.Set (Set)
+import Data.Typeable (Typeable)
+import GHC.Generics (Generic)
+import GHC.Records
+import NoThunks.Class (NoThunks (..))
+import Shelley.Spec.Ledger.BaseTypes (StrictMaybe)
+import Shelley.Spec.Ledger.Coin (Coin (..))
+import Shelley.Spec.Ledger.MetaData (MetaDataHash)
+import Shelley.Spec.Ledger.PParams (Update)
+import Shelley.Spec.Ledger.Serialization (encodeFoldable)
+import Shelley.Spec.Ledger.TxBody
+  ( DCert (..),
+    TxInParam (..),
+    TxOut (..),
+    Wdrl (..),
+  )
+
+-- ================================================
+-- There is a mutualy recursive cycle through the types
+-- TxId -> TxBody -> TxIn -> TxId.
+-- So we handle this by using the parameterized type
+-- (TxInParam (MaryBody era) era).  What this means that the
+-- TxOut's have TxId's can only have hashes of (MaryBody era)
+
+-- ===========================
+
+data MaryBody' era = MaryBody'
```

Comment:
Why did you rename this? It was much better when it was just `TxBody`. The era is implied by the module it's in

Reply:
Changed it back.